lib/charms/grafana_agent_k8s/v0/grafana_agent.py

```diff
@@ -97,9 +102,19 @@ def __init__(self, *args):
 
 DEFAULT_RELATION_NAME = "log_proxy"
 HTTP_LISTEN_PORT = 9080
+HTTP_LISTEN_PORT = 9080
 GRPC_LISTEN_PORT = 0
 POSITIONS_FILENAME = "/tmp/positions.yaml"
-CONFIG_PATH = "/etc/promtail/config.yml"
+CONFIG_PATH = "/tmp/promtail_config.yml"
+PROMTAIL_BINARY_ZIP_URL = "https://github.com/grafana/loki/releases/download/v2.4.1/promtail-linux-amd64.zip"
+BINARY_ZIP_PATH = "/tmp/promtail-linux-amd64.zip"
+BINARY_DIR = "/tmp"
+BINARY_SHA25SUM = "978391a174e71cfef444ab9dc012f95d5d7eae0d682eaf1da2ea18f793452031"
+WORKLOAD_BINARY_PATH = "/tmp/promtail-linux-amd64"
+
+
+class PromtailSHA256Error(Exception):
```

Comment:
Just call it `PromtailDigestError`, there is no reason to hard-code the hashing scheme in the exception name